### PR TITLE
Feat/init password vault

### PR DESCRIPTION
**Pull request by AbdulmajeedAlOmari**
FYR @AhmedAlzubaidi

### CODE REVIEW

File: Controllers/EncryptionKeyInfoController.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
+using ClassiVault.Api.DataAccess.Models;
+using Microsoft.AspNetCore.Mvc;
+
+[Route("api/[controller]")]
+[ApiController]
+public class EncryptionKeyInfoController : ControllerBase
```
Comment (AhmedAlzubaidi) [inline Controllers/EncryptionKeyInfoController.cs]: the route of this controller is /encryptionkeyinfo ?
if so can't we make it shorter

File: Controllers/PasswordVaultController.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
+using Microsoft.AspNetCore.Mvc;
+using ClassiVault.Api.DataAccess.Models;
+
+[Route("api/[controller]")]
```
Comment (AhmedAlzubaidi) [inline Controllers/PasswordVaultController.cs]: hmmm /vaults ? instead

File: DataAccess/AppDbContext.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        public DbSet<PasswordVault> PasswordVaults { get; set; }
+        public DbSet<EncryptionKeyInfo> EncryptionKeyInfos { get; set; }
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+            base.OnModelCreating(modelBuilder);
+
+            // Configure the primary keys
+            modelBuilder.Entity<PasswordVault>().HasKey(pv => pv.VaultID);
+            modelBuilder.Entity<EncryptionKeyInfo>().HasKey(eki => eki.KeyID);
```
Comment (AhmedAlzubaidi) [inline DataAccess/AppDbContext.cs]: here is what I suggest
- the primary key seems unpredictable change it to id
- all database keys lowercase
- underscore separated

File: DataAccess/Models/EncryptionKeyInfo.cs [hunk 4]
```diff
@@ -0,0 +1,30 @@
+
+namespace ClassiVault.Api.DataAccess.Models
+{
+  /// <summary>
+  /// The EncryptionKeyInfo model represents key-related information for each user.
+  /// This includes a unique identifier for the key, a reference to the user,
+  /// and a hash of the encryption key used for password encryption.
+  /// The key hash is stored for security purposes and does not expose the actual key.
+  /// </summary>
+  public class EncryptionKeyInfo
```
Comment (AhmedAlzubaidi) [inline DataAccess/Models/EncryptionKeyInfo.cs]: maybe shorten it to EncryptionKey

File: DataAccess/Models/EncryptionKeyInfo.cs [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  {
+      public long KeyID { get; set; }
+      public string UserID { get; set; } // Foreign Key
+      public string KeyHash { get; set; }
+      public DateTime CreatedAt { get; set; }
+      public DateTime UpdatedAt { get; set; }
```
Comment (AhmedAlzubaidi) [inline DataAccess/Models/EncryptionKeyInfo.cs]: I suggest we make a base model that contains CreatedAt and UpdatedAt
```c#
 public abstract class BaseEntity
 {
    public DateTime? CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
 }
 
 public class EncryptionKeyInfo : BaseEntity
```

File: DataAccess/Models/PasswordVault.cs [hunk 6]
```diff
@@ -0,0 +1,33 @@
+
+namespace ClassiVault.Api.DataAccess.Models
+{
+  /// <summary>
+  /// The PasswordVault model stores encrypted passwords and related information.
+  /// Each entry includes a unique identifier, a reference to the user,
+  /// the encrypted password, account identifier, and an optional website URL.
+  /// Timestamps for creation and last update are also maintained.
+  /// </summary>
+  public class PasswordVault
```
Comment (AhmedAlzubaidi) [inline DataAccess/Models/PasswordVault.cs]: I suggest we change this to Vault or ClassiVault 😄, because we might store other things that are not passwords

**Review by AhmedAlzubaidi — CHANGES_REQUESTED** [review]

File: Controllers/EncryptionKeyInfoController.cs (hunk 1 above)
Comment (AbdulmajeedAlOmari) [reply Controllers/EncryptionKeyInfoController.cs]: Done

File: Controllers/PasswordVaultController.cs (hunk 2 above)
Comment (AbdulmajeedAlOmari) [reply Controllers/PasswordVaultController.cs]: Done

File: DataAccess/AppDbContext.cs (hunk 3 above)
Comment (AbdulmajeedAlOmari) [reply DataAccess/AppDbContext.cs]: This is best practice naming convention for .NET apps, we can leave it as it is

Also, for primary keys,  we could go for both ways

Either rename all PKs to Id, or keep it as it is for better differentiation in queries later on

File: DataAccess/Models/EncryptionKeyInfo.cs (hunk 4 above)
Comment (AbdulmajeedAlOmari) [reply DataAccess/Models/EncryptionKeyInfo.cs]: Done

File: DataAccess/Models/EncryptionKeyInfo.cs (hunk 5 above)
Comment (AbdulmajeedAlOmari) [reply DataAccess/Models/EncryptionKeyInfo.cs]: Done

File: DataAccess/Models/PasswordVault.cs (hunk 6 above)
Comment (AbdulmajeedAlOmari) [reply DataAccess/Models/PasswordVault.cs]: Done

File: DataAccess/AppDbContext.cs (hunk 3 above)
Comment (AbdulmajeedAlOmari) [reply DataAccess/AppDbContext.cs]: Done

**Review by AhmedAlzubaidi — APPROVED** [review]

**Pull request merged**